cpp/findpath.cpp

```diff
@@ -109,6 +110,13 @@ bool MapSearchNode::IsSameState( MapSearchNode &rhs )
 
 }
 
+size_t MapSearchNode::Hash()

```

Comment:
What is the source of this hash?

Reply:
Seems like that's the example hash function. 
- https://en.cppreference.com/w/cpp/utility/hash
![image](https://user-images.githubusercontent.com/45742788/227534657-fdd8c874-d155-445b-bbda-4892d0d667a4.png)


The alternative hash_combine seems more involved.
- https://www.boost.org/doc/libs/1_64_0/boost/functional/hash/hash.hpp
- eventually calls `hash_combine_impl(..)`

Reply:
Jonathan is correct that it is pulled from the example on the hash documentation, and it worked well enough to keep. I did not want the library to require any extra boost library so I kept it simple. I'll make a note of where the hash came from.